docs/docs/file-system-page-creation.md

```diff
@@ -0,0 +1,173 @@
+---
+title: Creating Pages from the File System
+---
+
+This page roughly documents the APIs and conventions available with the new file-based page creation features.
+
+For now, these features are marked as experimental and require a flag to utilize. Start your Gatsby site locally with the following flag:
+
+```shell
+GATSBY_EXPERIMENTAL_ROUTING_APIS=1 gatsby develop
+```
+
+## Creating client-only routes
+
+Use client-only routes if you have dynamic data that does not live in Gatsby. This might be something like a user settings page, or some other dynamic content that isn't known to Gatsby at build time. In these situations, you will usually create a route with one or more dynamic segments to query data from a server in order to render your page.
+
+For example, in order to edit a user, you might want a route like `/user/:id` to fetch the data for whatever `id` is passed into the URL. You can now use square brackets (`[ ]`) in the file path to mark any dynamic segments of the URL.
+
+- `/src/pages/users/[id].js => /users/:id`
+- `/src/pages/users/[id]/group/[groupId].js => /users/:id/group/:groupId`
+
+Gatsby also supports _splat_ routes, which are routes that will match _anything_ after the splat. These are less common, but still have use cases. As an example, suppose that you are rendering images from [S3](/docs/deploying-to-s3-cloudfront/) and the URL is actually the key to the asset in AWS. Here is how you might create your file:
+
+- `/src/pages/image/[...awsKey].js => /users/*awsKey`
+- `/src/pages/image/[...].js => /app/*`
```

Comment:
the right side doesn't match left side here - on the left `/images/splat` - on the right `/users/splat` or `/app/splat`